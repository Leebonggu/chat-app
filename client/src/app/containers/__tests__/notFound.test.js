import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import NotFound from '../notFound';

const mockStore = configureStore([thunk]);

test('NotFound Container should render given props', () => {
  const initialState = {
    app: {
      isAuthenticated: false
    }
  };
  const store = mockStore(initialState);
  const mockFunc = jest.fn();
  const component = renderer.create(
    <Provider store={store}>
      <NotFound 
        goToLanding={mockFunc}
        goToLogin={mockFunc}
        isAuthenticated={false}
      />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
