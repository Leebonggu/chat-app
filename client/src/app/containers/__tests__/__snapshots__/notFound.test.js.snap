// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Container should render given props 1`] = `
<div
  className="notfound-container"
>
  <div
    className="text-exa"
  >
    404
  </div>
  <div
    className="text-tera"
  >
    There's Nothing here!
  </div>
  <div
    className="text-giga text-dim"
  >
    Not all wander are lost. But you certainly are.
  </div>
  <button
    className="secondary-button"
    onClick={[Function]}
    type="button"
  >
    Go back
  </button>
</div>
`;
